examples/dataservice-read/example.cpp

```diff
@@ -73,15 +73,27 @@ int main() {
           catalogResponse.GetResult();
       LOG_INFO_F("read-example", "Catalog description: %s",
                  responseResult.GetDescription().c_str());
+      auto layers = catalogResponse.GetResult().GetLayers();
+      if (!layers.empty()) {
+        firstPartitionId = layers.front().GetId();
+      }
+      auto end = layers.size() <= gMaxLayers ? layers.end()
+                                             : layers.begin() + gMaxLayers;
+      for (auto it = layers.cbegin(); it != end; ++it) {
+        LOG_INFO_F("read-example", "Layer '%s' (%s): %s", it->GetId().c_str(),
+                   it->GetLayerType().c_str(), it->GetDescription().c_str());
+      }
     } else {
       LOG_ERROR("read-example", "Request catalog metadata - Failure");
     }
   }
 
-  {  // Retrieve the partitions metadata
+  std::string firstLayerId;
+  if (!firstPartitionId.empty()) {
```

Comment:
no else statement, the example will not show anything if something wrong happened

Reply:
added a warning that there are no partitions or layers